packages/adapter/src/response-processing/casting.ts

```diff
@@ -53,25 +54,56 @@ function castBoolean(value: unknown): boolean {
   }
 }
 
-function castBytes32(value: any): string {
+function castStringLike(value: any, target: 'string' | 'bytes32'): string {
+  // TODO: Should we throw here? Isn't String(JSON.stringify(value)) better?
   // Objects convert to "[object Object]" which isn't very useful
   if (isArray(value) || isPlainObject(value)) {
-    throw new Error(`Unable to convert: '${JSON.stringify(value)}' to bytes32`);
+    throw new Error(`Unable to convert: '${JSON.stringify(value)}' to ${target}`);
   }
```

Comment:
This now looks like a good candidate for a separate function since we are duplicating these lines in 3 different places now.